src/pact.ts

```diff
@@ -56,6 +60,48 @@ export class Pact {
 		return q.all<Server>(_.map(this.__servers, (server:Server) => server.delete() as PromiseLike<Server>));
 	}
 
+	// Creates stub with specified options
+	public createStub(options: StubOptions = {}): Stub {
+		if (options && options.port && _.some(this.__stubs, (s: Stub) => s.options.port === options.port)) {
+			let msg = `Port '${options.port}' is already in use by another process.`;
+			logger.error(msg);
+			throw new Error(msg);
+		}
+
+		let stub = stubFactory(options);
+		this.__stubs.push(stub);
+		logger.info(`Creating Pact Stub with options: \n${this.__stringifyOptions(stub.options)}`);
+
+		// Listen to stub delete events, to remove from stub list
+		stub.once(AbstractService.Events.DELETE_EVENT, (s: Stub) => {
+			logger.info(`Deleting Pact Stub with options: \n${this.__stringifyOptions(s.options)}`);
+			this.__stubs = _.without(this.__stubs, s);
+		});
+
+		return stub;
+	}
+
+	// Return arrays of all stubs
+	public listStubs(): Stub[] {
+		return this.__stubs;
+	}
+
+	// Remove all the stubs that have been created
+	// Return promise of all others
+	public removeAllStubs(): q.Promise<Stub[]> {
+		if(this.__stubs.length === 0) {
+			return q(this.__stubs);
+		}
+
+		logger.info("Removing all Pact stubs.");
+		return q.all<Stub>(_.map(this.__stubs, (stub:Stub) => stub.delete() as PromiseLike<Stub>));
+	}
+
+	// Remove all the servers and stubs
+	public removeAll(): q.Promise<any> {
```

Comment:
Should the return type be a promise with AbstractService?

Reply:
Arguably, but the compiler didn't agree with me. I thought given both Stub and Server were subtypes this would work - but alas, no. 

I could create an interface so that it should work given its structural type, but that felt even worse.

Ideas?